File: server/go/unsubscribe-user-from-list.go
//go:build ignore

// +build ignore

package main

import (
	"context"
	"fmt"
	"os"
	"path/filepath"

	"github.com/joho/godotenv"
	"github.com/trycourier/courier-go/v4"
	"github.com/trycourier/courier-go/v4/option"
)

func LoadEnv() {
	envPath := filepath.Join("..", ".env")
	godotenv.Load(envPath)
}

func main() {
	LoadEnv()

	apiKey := os.Getenv("COURIER_API_KEY")
	listID := os.Getenv("COURIER_UNSUBSCRIBE_USER_FROM_LIST_LIST_ID")
	userID := os.Getenv("COURIER_UNSUBSCRIBE_USER_FROM_LIST_USER_ID")

	client := courier.NewClient(
		option.WithAPIKey(apiKey),
	)

	var response map[string]interface{}
	err := client.Delete(
		context.Background(),
		fmt.Sprintf("/lists/%s/subscriptions/%s", listID, userID),
		nil,
		&response,
	)

	if err != nil {
		panic(err.Error())
	}

	fmt.Printf("%+v\n", response)
}

File: server/ruby/unsubscribe-user-from-list.rb
require 'dotenv'
require 'courier'

env_path = File.join(File.dirname(__FILE__), '..', '.env')
Dotenv.load(env_path)

api_key = ENV['COURIER_API_KEY']
list_id = ENV['COURIER_UNSUBSCRIBE_USER_FROM_LIST_LIST_ID']
user_id = ENV['COURIER_UNSUBSCRIBE_USER_FROM_LIST_USER_ID']

client = Courier::Client.new(api_key: api_key)

client.lists.subscriptions.unsubscribe_user(user_id, list_id: list_id)
